test/serializers/attribute_test.rb

```diff
@@ -71,6 +71,21 @@ def id
 
         assert_equal('custom', hash[:blog][:id])
       end
+
+      PostWithVirtualAttribute = Class.new(::Model)
+      class PostWithVirtualAttributeSerializer < ActiveModel::Serializer
+        attribute :name do
+          "#{object.first_name} #{object.last_name}"
+        end
+      end
```

Comment:
@noahsilas I do want to remove the dynamic methods on the serializer, just not in this PR.

However, I do think it's a good idea for there to be a test against attribute names/keys that conflict with methods on the serializer.  I think the `read_attribute_for_serialization` should get around that